crates/ursa/src/config.rs

```diff
@@ -1,26 +1,33 @@
+use anyhow::{anyhow, Result};
 use serde::{Deserialize, Serialize};
+use std::{
+    fs::{create_dir_all, File},
+    io::{Read, Write},
+    path::{Path, PathBuf},
+    process,
+};
+use tracing::{error, info};
 use ursa_index_provider::config::ProviderConfig;
 use ursa_metrics::config::MetricsServiceConfig;
 use ursa_network::NetworkConfig;
 use ursa_rpc_server::config::ServerConfig;
 
-use std::{
-    fs::{create_dir_all, File},
-    io::{Result, Write},
-    path::PathBuf,
-};
-
 pub const DEFAULT_CONFIG_PATH_STR: &str = ".ursa/config.toml";
 
-pub fn load_config(path: &PathBuf) -> Result<()> {
-    if !path.exists() {
+pub fn load_config(path: &PathBuf) -> Result<UrsaConfig> {
+    info!("Loading config from: {:?}", path);
+    if path.exists() {
+        let toml = read_file_to_string(path)?;
+        toml::from_str(&toml).map_err(|e| anyhow!("Failed to parse config file: {}", e))
+    } else {
+        // Missing, create and return default config at path
         let ursa_config = UrsaConfig::default();
         let toml = toml::to_string(&ursa_config).unwrap();
         create_dir_all(path.parent().unwrap())?;
         let mut file = File::create(path)?;
-        return file.write_all(toml.as_bytes());
+        file.write_all(toml.as_bytes())?;
```

Comment:
use `.context()` to give a little context to the error messages for better diagnostics for users.

Reply:
https://github.com/fleek-network/ursa/pull/116/commits/7e776b95681a5a63c6776418a235945b9f27fb16